src/Runner.Worker/ActionRunner.cs

```diff
@@ -355,7 +355,7 @@ public bool EvaluateDisplayName(DictionaryContextData contextData, IExecutionCon
         {
             DictionaryContextData expressionValues = ExecutionContext.GetExpressionValues(stepHost);
             var templateEvaluator = ExecutionContext.ToPipelineTemplateEvaluator();
-            var inputs = templateEvaluator.EvaluateStepInputs(Action.Inputs, expressionValues, ExecutionContext.ExpressionFunctions);
+            var inputs = templateEvaluator.EvaluateStepInputs(Action.Inputs, expressionValues, ExecutionContext.ExpressionFunctions, ExecutionContext.ToExpressionState());
```

Comment:
Passing down ExecutionContext (like we do for StepIf, like `success()` makes it available in the function later)